protocols/floodsub/src/protocol.rs

```diff
@@ -93,21 +106,24 @@ where
 
 /// Reach attempt interrupt errors.
 #[derive(thiserror::Error, Debug)]
-pub enum FloodsubDecodeError {
-    /// Error when reading the packet from the socket.
-    #[error("Failed to read from socket")]
-    ReadError(#[from] io::Error),
-    /// Error when decoding the raw buffer into a protobuf.
-    #[error("Failed to decode protobuf")]
-    ProtobufError(#[from] DecodeError),
+pub enum FloodsubError {
+    /// Error from prost_codec::Codec
+    #[error(transparent)]
+    Codec(#[from] prost_codec::Error),
```

Comment:
This one should not exist! We want to hide `prost_codec::Error` from the public API of this crate!

Reply:
That is the reason for `CodecError` existing :)

Reply:
umm, I just got side tracked on this. While trying to fix it I caused an error ` the compiler unexpectedly panicked. this is a bug.` I'm going to go report that and then pick this up again tomorrow. Thanks for all your help on this @thomaseizinger!

Reply:
> umm, I just got side tracked on this. While trying to fix it I caused an error ` the compiler unexpectedly panicked. this is a bug.` I'm going to go report that and then pick this up again tomorrow. Thanks for all your help on this @thomaseizinger!

`cargo clean` usually fixes these. I hit them too occasionally, often in combination with proc-macros.

Reply:
That did the trick! I cleaned up the commit message and pushed what I think is the corrected code.